### PR TITLE
Add pythonw.exe to Windows CPython test runtimes.

### LINKED ISSUE #238: Test runtimes should include pythonw.exe
This is needed for testing functions that use pythonw.exe rather than python.exe

### PR DESCRIPTION
**Pull request by JCorson**
Fixes #238. 
This adds pythonw.exe to the test Windows runtimes. This is needed for tests to pass in edm with the changes needed for https://github.com/enthought/edm/issues/1081

### CODE REVIEW

File: okonomiyaki/utils/tests/test_test_data.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import sys
+import zipfile2
+
+from .. import test_data
+
+if sys.version_info < (2, 7):
+        import unittest2 as unittest
```
Comment (cournape) [inline okonomiyaki/utils/tests/test_test_data.py]: indentation

File: okonomiyaki/utils/tests/test_test_data.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        with zipfile2.ZipFile(runtime) as zp:
+            return zp.namelist()
+
+    def test_pythonw_in_dummy_runtime(self):
+        """
+        Ensure that pythonw.exe is included in all Windows runtimes
+        """
+
+        # Given
+        win_runtimes = [f for f in dir(test_data)
```
Comment (cournape) [inline okonomiyaki/utils/tests/test_test_data.py]: This is not the right way of detecting implementation/platform: you should instead use `okonomiyaki.runtimes.IRuntimeMetadata.factory_from_path`, which gives you an object w/ both a platform and implementation attributes.

Comment (cournape) [inline okonomiyaki/utils/tests/test_test_data.py]: My suggestion here would be either being more explicit (just hardcoding the list of runtimes here), or have a special variable in `okonomiyaki.utils.test_data` that list all runtimes and do the filtering based on `IRuntimeMetadata.factory_from_path`.

File: okonomiyaki/utils/tests/test_test_data.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        with zipfile2.ZipFile(runtime) as zp:
+            return zp.namelist()
+
+    def test_pythonw_in_dummy_runtime(self):
+        """
+        Ensure that pythonw.exe is included in all Windows runtimes
+        """
+
+        # Given
+        python_runtimes = [attribute for attribute in dir(test_data)
```
Comment (cournape) [inline okonomiyaki/utils/tests/test_test_data.py]: I am still a bit worried about relying on the variable name. If you really want to keep the "automatic discovery" magic, I would advise to just get all the attributes and filter it out where it ends with a `.runtime`:

``` python
test_data_vars = [attribute for attribute in dir(test_data) if getattr(test_data, attribute).endswith(".runtime")]
...
```

File: okonomiyaki/utils/tests/test_test_data.py (hunk 1 above)
Comment (JCorson) [inline okonomiyaki/utils/tests/test_test_data.py]: Fixed with https://github.com/enthought/okonomiyaki/pull/239/commits/4ba6b06e94f64ad6b86f4ab973c77d0d0970c702

**Comment by JCorson** [comment]
I think this is ready for another review @cournape. I would like to keep the automatic detection of runtimes because it will ensure that future runtimes are not missed in the testing.  I also opted to just search the test_data path for runtimes rather than checking all of the attributes in the test_data module. Seems simpler and more straightforward.

File: okonomiyaki/utils/tests/test_test_data.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        with zipfile2.ZipFile(runtime) as zp:
+            return zp.namelist()
+
+    def test_pythonw_in_dummy_runtime(self):
+        """
+        Ensure that pythonw.exe is included in all Windows runtimes
+        """
+
+        # Given
+        test_data_dir = join(dirname(__file__), '..', 'test_data')
```
Comment (cournape) [inline okonomiyaki/utils/tests/test_test_data.py]: Can we instead have a `DUMMY_RUNTIMES_DIRECTORY` in okonomiyaki.utils.test_data that points to this instead ?

**Comment by cournape** [comment]
@JCorson one last comment, and then feel free to merge.

**Comment by JCorson** [comment]
All tests passing. Merging.

**Pull request merged**